docs/what-just-happened/what-just-happened_test_plan.md

```diff
@@ -0,0 +1,145 @@
+# what-just-happened Test Plan
+
+## Table of Contents
+- [Introduction](#introduction)
+- [Pre test](#pre-test)
+- [Test Plan](#test-plan)
+    - [Overview](#overview)
+    - [Test Cases](#test-cases)
+        - [L2 Drops](#l2-drops)
+        - [L3 Drops](#l3-drops)
+
+
+## Introduction
+what-just-happened is a feature intended to track lost packets and display the drop reason.
+
+The purpose of this test is:
+
+1. Check that wrong packets are dropped. 
+2. Check that the description in 'show what-just-happened' command output suits the dropped packet.
+ 
+The test will focus on L2, L3 raw packets drops. 
+
+The implementation will be done using PyTest framework.
+
+## Pre test
+Before all test cases execution, the test should check whether what-just-happened feature is available and enabled. 
+
+For this purpose, a pre test fixture will do the following:
+
+- Execute ```show feature``` command on DUT. 
+- Parse output table.
+- Check if 'what-just-happened' appears in the table (availability), and if it is enabled.
+
+```bash
+Feature             Status
+------------------  --------
+telemetry           enabled
+sflow               disabled
+what-just-happened  enabled
+```
+
+- If one of the checks on previous phase is indicated what-just-happened as not available, skip the test. 
```

Comment:
You probably need to verify that debug mode is enabled by looking at "WJH" table and verify or set channels in WJH_CHANNEL. For forwarding - you need to ensure that the relevant channel exists in DB.